File: case_studies/m/auv_depth.m
function xdot = dynamics(t, x, u)
    m = 50; Iz = 5; W = 500; B = 510; zb = -0.1; zg = 0;
    Zw = -20; Zq = -5; Zd = -10; Mw = -3; Mq = -8; Md = -5;
    z = x(1); w = x(2); theta = x(3); q = x(4);
    xdot = [w*cos(theta) - sin(theta);
            (Zw*w + Zq*q + Zd*u + (W-B)*sin(theta)) / m;
            q;
            (Mw*w + Mq*q + Md*u + (zg*W - zb*B)*sin(theta)) / Iz];
end

File: case_studies/py/auv_depth.py
import numpy as np

def dynamics(t, x, u):
    m, Iz, W, B, zb, zg = 50, 5, 500, 510, -0.1, 0
    Zw, Zq, Zd, Mw, Mq, Md = -20, -5, -10, -3, -8, -5
    z, w, theta, q = x[0], x[1], x[2], x[3]
    xdot = [w*np.cos(theta) - np.sin(theta),
            (Zw*w + Zq*q + Zd*u + (W-B)*np.sin(theta)) / m,
            q,
            (Mw*w + Mq*q + Md*u + (zg*W - zb*B)*np.sin(theta)) / Iz]
    return np.array(xdot)